src/fsharp/DependencyManager/NativeDllResolveHandler.fs

```diff
@@ -111,25 +111,31 @@ type NativeDllResolveHandler (nativeProbingRoots: NativeResolutionProbe) =
         else
             None
 
-    let appendSemiColon (p: string) =
-        if not(p.EndsWith(";", StringComparison.OrdinalIgnoreCase)) then
-            p + ";"
+    let pathSeparator =
+        if RuntimeInformation.IsOSPlatform(OSPlatform.Windows) then
+            ";"
+        else
+            ":"
+
+    let appendPathSeparator (p: string) =
+        if not(p.EndsWith(pathSeparator, StringComparison.OrdinalIgnoreCase)) then
+            p + pathSeparator
```

Comment:
Use `System.IO.Path.PathSeparator` and drop the `pathSeparator` value, otherwise this is great, thank you.

Reply:
wow, I even said to do that in my comment in the original issue 🤦